app/models/user.rb

```diff
@@ -0,0 +1,6 @@
+class User < ApplicationRecord
+    has_many :posts
+    has_many :comments
+    validates :name, presence: true, length: { minimum: 4, maximum: 50 }
+    validates :email, presence: true, length: { minimum: 8, maximum: 250 }
```

Comment:
Same here, this allows the creation users with the same email:

```
irb(main):008:0> User.where(email: 'johndoe@example.com').count
   (0.4ms)  SELECT COUNT(*) FROM "users" WHERE "users"."email" = ?  [["email", "johndoe@example.com"]]
=> 2
```

Furthermore, validating the email against a regular expression pattern might also be something to consider. One can find examples online of a valid regex, but here's one that I've used:

`VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i`

[OPTIONAL]